pkg/kubelet/kubeletconfig/configsync.go

```diff
@@ -181,3 +205,43 @@ func latestNode(store cache.Store, nodeName string) (*apiv1.Node, error) {
 	}
 	return node, nil
 }
+
+// eventf constructs and returns an event containing a formatted message
+// similar to k8s.io/client-go/tools/record/event.go
+func eventf(nodeName, eventtype, reason, messageFmt string, args ...interface{}) *apiv1.Event {
+	return makeEvent(nodeName, eventtype, reason, fmt.Sprintf(messageFmt, args...))
+}
+
+// makeEvent constructs an event
+// similar to makeEvent in k8s.io/client-go/tools/record/event.go
+func makeEvent(nodeName, eventtype, reason, message string) *apiv1.Event {
+	const componentKubelet = "kubelet"
+	// NOTE(mtaufen): This is consistent with pkg/kubelet/kubelet.go. Even though setting the node
+	// name as the UID looks strange, it appears to be conventional for events sent by the Kubelet.
```

Comment:
Hmm..why is that? Is it because kubelet may send an event before it registers itself?

Reply:
That sounds reasonable, though I don't know if that's the actual reason.
The node name is _technically_ a unique id for the Kubelet, so I guess it works in a way.
In any case, it's been the convention in the Kubelet, and we should stick to it without a separate (probably large) discussion.